activation/activation.go

```diff
@@ -19,41 +21,67 @@ type MeshProvider interface {
 	LatestLayerId() mesh.LayerID
 }
 
+type EpochId uint64
+
+func (l EpochId) ToBytes() []byte { return common.Uint64ToBytes(uint64(l)) }
+
+type EpochProvider interface {
+	Epoch(l mesh.LayerID) EpochId
+}
+
 type Builder struct {
-	nodeId    mesh.NodeId
-	db        *ActivationDb
-	net       api.NetworkAPI
-	activeSet ActiveSetProvider
-	mesh      MeshProvider
+	nodeId        mesh.NodeId
+	db            *ActivationDb
+	net           api.NetworkAPI
+	activeSet     ActiveSetProvider
+	mesh          MeshProvider
+	epochProvider EpochProvider
 }
 
 type Processor struct {
-	db *ActivationDb
+	db            *ActivationDb
+	epochProvider EpochProvider
 }
 
 func (p *Processor) ProcessBlockATXs(block *mesh.Block) {
 	for _, atx := range block.ATXs {
-		p.db.StoreAtx(atx)
+		err := p.db.StoreAtx(p.epochProvider.Epoch(atx.LayerIndex), atx)
+		if err != nil {
+			log.Error("cannot store atx: %v", atx)
+		}
 	}
 }
 
-func NewBuilder(nodeId mesh.NodeId, db database.DB, net api.NetworkAPI, activeSet ActiveSetProvider, view MeshProvider) *Builder {
+func NewBuilder(nodeId mesh.NodeId, db database.DB, net api.NetworkAPI, activeSet ActiveSetProvider, view MeshProvider, epochDuration EpochProvider) *Builder {
 	return &Builder{
-		nodeId, &ActivationDb{db}, net, activeSet, view,
+		nodeId, &ActivationDb{db}, net, activeSet, view, epochDuration,
 	}
 }
 
-func (b Builder) BuildActivationTx(nipst *nipst.NIPST) error {
+func (b *Builder) BuildActivationTx(nipst *nipst.NIPST) error {
```

Comment:
it's kinda weird that BuildActivationTx broadcasts the tx. I would expect the name to disclose that.